# (PART\*) Genomic selection of sources  
# Setting up {-} 

Read in the necessary data for the source optimization. [SnpEff v5.1](https://pcingola.github.io/SnpEff/) *(Cingolani et al., 2012)* was used to annotate genetic variants to functional class based on Norway spruce genome annotation. 


## Import annotations from SnpEff {-} 
```{r eval=FALSE}
## working directory 
setwd("~/mydata/Anoob/TNC")

list <- list.files(path = "../../datashare/Spruce/exome_capture/WES_mapping/Annotations/ref_Pglauca/VCF_split_files", 
                   pattern = "Red_Spruce_intersect_poly_", recursive=TRUE, full.names = T)

genes <- lapply(list[1], function(x) read.table(x))
category <- lapply(genes, function(x) unlist(lapply(strsplit(as.character(x[,8]), split = "|", fixed = T), function(y) y[2])))

TAB <- genes[1:2]
TAB <- do.call(rbind, TAB)
category <- do.call(c, category)
```

Read in the meta data for the samples.   
```{r eval=FALSE}
# Info samples
names <- unlist(lapply(strsplit(unlist(strsplit(as.character(read.table("all_bam.list")[,1]), split = "_rmdup.sorted.bam")), split = "./"), function(x) x[2]))
pops <- unlist(lapply(strsplit(names, split="_"), function(x) x[1]))

info_inds <- read.table("./Info_samples_revised.txt", header=T)
info_inds <- info_inds[match(as.character(names), as.character(info_inds$Family)),]
info_pops <- info_inds[!duplicated(info_inds$Site),-c(1,3,9,10)]
```

## Allele probabilities and frequencies   {-} 
```{r eval=FALSE}
# Read depth 
depth <- apply(TAB[,-c(1:9)], 2, function(x) as.integer(unlist(lapply(strsplit(as.character(x), split = ":"), function(y) y[2]))))

# Genotype probabilities, changed by NA for the uncovered sites 
gen_prob <- apply(TAB[,-c(1:9)], 2, function(x) unlist(lapply(strsplit(as.character(x), split = ":"), function(y) y[4])))
gen_prob[which(depth==0)] <- NA

# Proba alternative allele
altern_proba <- apply(gen_prob, 2, function(x) (as.numeric(unlist(lapply(strsplit(as.character(x), split = ","), function(y) y[2])))+2*as.numeric(unlist(lapply(strsplit(as.character(x), split = ","), function(z) z[3]))))/2)

# Frequency of the alternative allele for each locus and population
TAB_pop <- lapply(unique(pops), function(x) altern_proba[,which(pops==x)])
names(TAB_pop) <- unique(pops)
freq_pop <- lapply(TAB_pop, function(x) apply(x, 1, function(y) sum(y, na.rm = T)/sum(!is.na(y))))
```